tests/instron_stress_rig.py

```diff
@@ -325,5 +355,237 @@ def test_WHEN_the_channel_type_updates_on_the_device_THEN_the_pv_updates(self):
+        expected_values = [123.456, 789.012, 345.678]
+        assert len(expected_values) == NUMBER_OF_CHANNELS
+
+        for channel in range(NUMBER_OF_CHANNELS):
+            self.ca.set_pv_value("CHANNEL:SP.VAL", channel)
+            self.ca.set_pv_value(wave_prefixed("FREQ:SP"), expected_values[channel])
+        for channel in range(NUMBER_OF_CHANNELS):
+            self.ca.set_pv_value("CHANNEL:SP.VAL", channel)
+            self.ca.assert_that_pv_is(wave_prefixed("FREQ"), expected_values[channel])
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails") # Conversion factors initialized to 0
+    def test_GIVEN_multiple_channels_WHEN_waveform_amplitude_is_set_THEN_the_device_is_updated_to_that_value_with_channel_conversion_factor_applied(self):
+        input_values = [123.456, 789.012, 345.678]
+        conversion_factors = [
+            float(self.ca.get_pv_value("POS:SCALE"))*1000.0,
+            float(self.ca.get_pv_value("STRESS:SCALE"))*float(self.ca.get_pv_value("STRESS:AREA")),
+            float(self.ca.get_pv_value("STRAIN:SCALE"))*100000.0/float(self.ca.get_pv_value("STRAIN:LENGTH"))
+        ]
+        expected_values = [input_values[i]/conversion_factors[i] for i in range(NUMBER_OF_CHANNELS)]
+        assert len(expected_values) == len(conversion_factors) == len(input_values) == NUMBER_OF_CHANNELS
+
+        for channel in range(NUMBER_OF_CHANNELS):
+            self.ca.set_pv_value("CHANNEL:SP.VAL", channel)
+            self.ca.set_pv_value(wave_prefixed("AMP:SP"), input_values[channel])
+        for channel in range(NUMBER_OF_CHANNELS):
+            self.ca.set_pv_value("CHANNEL:SP.VAL", channel)
+            self.ca.assert_that_pv_is_number(wave_prefixed("AMP"), expected_values[channel], tolerance=0.0005)
+
+    def test_WHEN_the_quarter_counter_is_off_THEN_the_number_of_counts_is_and_remains_zero(self):
+        self.ca.set_pv_value(quart_prefixed("OFF"), 1)
+        self.ca.assert_that_pv_is("QUART", 0)
+        sleep(5)
+        self.ca.assert_that_pv_is("QUART", 0)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails") # Status more complicated than RECSIM can handle
+    def test_WHEN_the_quarter_counter_is_armed_THEN_the_status_is_armed(self):
+        self.ca.set_pv_value(quart_prefixed("ARM"), 1)
+        self.ca.assert_that_pv_is(quart_prefixed("STATUS"), "Armed")
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails") # Counting part of dynamic device behaviour
+    def test_WHEN_the_waveform_generator_is_started_THEN_the_quarter_counter_starts_counting_and_keeps_increasing(self):
+        self.ca.set_pv_value(quart_prefixed("CYCLE:SP"), LOTS_OF_CYCLES)
+        self.ca.set_pv_value(wave_prefixed("START"), 1)
+        first_count = float(self.ca.get_pv_value("QUART"))
+        sleep(5)
+        second_count = float(self.ca.get_pv_value("QUART"))
+        self.assertGreater(second_count, first_count)
```

Comment:
I've added this function as well as decreasing and unchanged. There is also a generic function that accepts a comparator.